cypress/e2e/editProfile.spec.ts

```diff
@@ -0,0 +1,33 @@
+/// <reference types="cypress" />
+import '../support/commands';
+
+describe('Edit profile functions correctly', () => {
+  before(() => {
+    cy.createUsers(10);
+  });
+
+  beforeEach(() => {
+    cy.login(1).reload();
+    cy.createUsers(10);
+  });
+
+  it('Prevents a user from editing other users profiles', () => {
+    cy.visit('http://localhost:3000/profile/2');
+    cy.contains('button', 'Edit Profile').should('not.exist');
```

Comment:
This isn't blocking, but instead of selecting elements by `button` and `'textarea[name=bio]'` further down, which may be clear now before there is nothing else on this page. Might not be clear in the future. We can future proof this by adding data id's to hook into for the E2E test

example: 
```tsx
   <Button
              variant="contained"
              color="primary"
              className="text-2xl font-700 text-center"
              onClick={handleToggleForm}
              data-cy="edit-button"
            >
              Edit Profile
            </Button>
```

Then select it like so
```ts
cy.get('[data-cy="edit-button"]')
```

This way is abundantly clear a test is attached to that element and you can search by the `data-cy="edit-button"` to see what tests are looking for the element so you know which ones to update in the future